cli/cmd/edges.go

```diff
@@ -0,0 +1,360 @@
+
+func writeEdgesToBuffer(rows []*pb.Edge, w *tabwriter.Writer, options *edgesOptions) {
+	maxSrcLength := len(srcHeader)
+	maxDstLength := len(dstHeader)
+	maxClientLength := len(clientHeader)
+	maxServerLength := len(serverHeader)
+	maxMsgLength := len(msgHeader)
+
+	edgeTables := make(map[string]*edgeRow)
+	if len(rows) != 0 {
+		for _, r := range rows {
+			key := r.Dst.Name + r.Src.Name
+			clientID := r.ClientId
+			serverID := r.ServerId
+			msg := r.NoIdentityMsg
+
+			if len(msg) == 0 {
+				msg = "-"
+			}
+			if len(clientID) > 0 {
+				parts := strings.Split(clientID, ".")
+				clientID = parts[0] + "." + parts[1]
+			}
+			if len(serverID) > 0 {
+				parts := strings.Split(serverID, ".")
+				serverID = parts[0] + "." + parts[1]
+			}
+
+			edgeTables[key] = &edgeRow{
+				client: clientID,
+				server: serverID,
+				msg:    msg,
+				src:    r.Src.Name,
+				dst:    r.Dst.Name,
+			}
+
+			if len(r.Src.Name) > maxSrcLength {
+				maxSrcLength = len(r.Src.Name)
+			}
+			if len(r.Dst.Name) > maxDstLength {
+				maxDstLength = len(r.Dst.Name)
+			}
+			if len(clientID) > maxClientLength {
+				maxClientLength = len(clientID)
+			}
+			if len(serverID) > maxServerLength {
+				maxServerLength = len(serverID)
+			}
+			if len(msg) > maxMsgLength {
+				maxMsgLength = len(msg)
+			}
+
+		}
+	}
+	switch options.outputFormat {
+	case tableOutput:
+		if len(edgeTables) == 0 {
+			fmt.Fprintln(os.Stderr, "No edges found.")
+			os.Exit(0)
+		}
+		printEdgeTable(edgeTables, w, maxSrcLength, maxDstLength, maxClientLength, maxServerLength, maxMsgLength)
+	case jsonOutput:
+		printEdgesJSON(edgeTables, w)
+	}
+}
+
+func printEdgeTable(edges map[string]*edgeRow, w *tabwriter.Writer, maxSrcLength, maxDstLength, maxClientLength, maxServerLength, maxMsgLength int) {
+	srcTemplate := fmt.Sprintf("%%-%ds", maxSrcLength)
+	dstTemplate := fmt.Sprintf("%%-%ds", maxDstLength)
+	clientTemplate := fmt.Sprintf("%%-%ds", maxClientLength)
+	serverTemplate := fmt.Sprintf("%%-%ds", maxServerLength)
+	msgTemplate := fmt.Sprintf("%%-%ds", maxMsgLength)
+
+	headers := []string{
+		fmt.Sprintf(srcTemplate, srcHeader),
+		fmt.Sprintf(dstTemplate, dstHeader),
+		fmt.Sprintf(clientTemplate, clientHeader),
+		fmt.Sprintf(serverTemplate, serverHeader),
+		fmt.Sprintf(msgTemplate, msgHeader),
+	}
+
+	headers[len(headers)-1] = headers[len(headers)-1] + "\t" // trailing \t is required to format last column
+
+	fmt.Fprintln(w, strings.Join(headers, "\t"))
+
+	sortedKeys := sortEdgesKeys(edges)
+	for _, key := range sortedKeys {
+		values := make([]interface{}, 0)
+		templateString := fmt.Sprintf("%s\t%s\t%s\t%s\t%s\t\n", srcTemplate, dstTemplate, clientTemplate, serverTemplate, msgTemplate)
+
+		values = append(values, []interface{}{
+			edges[key].src,
+			edges[key].dst,
+			edges[key].client,
+			edges[key].server,
+			edges[key].msg,
+		}...)
+
+		fmt.Fprintf(w, templateString, values...)
+
+	}
+}
+
+func renderEdges(buffer bytes.Buffer, options *edgesOptions) string {
```

Comment:
tioli: consider re-using `renderStats`.

it would require changing its signature to:
```go
func renderStats(buffer bytes.Buffer, outputFormat string) string {
```
... but i think that's a good change to make anyway.

Reply:
Great idea. Flagging for a separate PR.